silverstripe-terraformers/keys-for-cache
# Add support for Silverstripe 5

Fixes #69
Closes #69

## New MAJOR

This release will be tagged as version `2.0.0`.

I have created a new branch `1` for continued support for Silverstripe 4.

## Gridfield Orderable Rows workaround

No longer needed, as the minimum supported version of Framework now includes the original bugfix.

## PHP ^8 goodies

* [PHP 8.0: Class constructor property promotion](https://php.watch/versions/8.0/constructor-property-promotion)
* [PHP 8.1: Readonly Properties](https://php.watch/versions/8.1/readonly)

## Manual testing

Given our test coverage, I felt I really only needed to perform some basic functional testing.

* Page created with `has_cache_key` and a `cares` for an `Image`
* Output `$Page.CacheKey` into a template so that I can see the current value
* Visit Page in mode `Stage`, note the key
* Visit Page in mode `Live`, note the key
* Edit the Page record itself
* Save the Page (don't publish)
* Confirmed: Key in mode `Stage` changed
* Confirmed: Key in mode `Live` did not change
* Publish the Page
* Confirmed: Key in mode `Live` changed
* Edit the Image that is assigned to the page
* Save the Image (don't publish)
* Confirmed: Key in mode `Stage` changed
* Confirmed: Key in mode `Live` did not change
* Publish the Image
* Confirmed: Key in mode `Live` changed
